apps/scan/frontend/test/helpers/mock_api_client.tsx

```diff
@@ -50,13 +52,70 @@ export const statusNoPaper: PrecinctScannerStatus = {
   ballotsCounted: 0,
 };
 
+/**
+ * Because you can get to the opened state by either opening polls or resuming
+ * voting, we don't know exactly what the last transition was. But we want to
+ * interpolate our best guess for testing ease.
+ */
+function getLikelyLastPollsTransitionType(
+  pollsState: Exclude<PollsState, 'polls_closed_initial'>
+): PollsTransitionType {
+  switch (pollsState) {
+    case 'polls_closed_final':
+      return 'close_polls';
+    case 'polls_open':
+      return 'open_polls';
+    case 'polls_paused':
+      return 'pause_voting';
+    // istanbul ignore next
+    default:
+      throwIllegalValue(pollsState);
+  }
+}
+
+export function mockPollsInfo(
+  pollsState: PollsState,
+  lastPollsTransition?: Partial<PollsTransition>
+): PrecinctScannerPollsInfo {
+  if (pollsState === 'polls_closed_initial') {
+    return {
+      pollsState,
+    };
+  }
+  return {
+    pollsState,
+    lastPollsTransition: {
+      ...(lastPollsTransition ?? {}),
+      type: getLikelyLastPollsTransitionType(pollsState),
+      time: Date.now(),
+      ballotCount: 0,
+    },
+  };
+}
+
+type MockApiClient = Omit<MockClient<Api>, 'transitionPolls'> & {
+  // Because transitionPolls takes a timestamp as an argument, which is difficult to mock
+  // precisely even with fake timers, we use a jest mock for more flexible matching
+  transitionPolls: jest.Mock;
+};
+
+function createMockApiClient(): MockApiClient {
+  const mockApiClient = createMockClient<Api>();
+  // For some reason, using an object spread to override methods breaks the rest
+  // of the mockApiClient, so we override like this instead
+  (mockApiClient.transitionPolls as unknown as jest.Mock) = jest.fn(() => {
+    throw new Error('transitionPolls not mocked');
+  });
+  return mockApiClient as unknown as MockApiClient;
+}
+
```

Comment:
I tried to avoid introducing something like this, but I couldn't correctly mock the timestamp that is required on transitioning polls. Even if you mock the system time via `jest`, it will mock advancing time throughout the test so the actual timestamp will have drifted from the initial mock.

Reply:
One workaround might be to have a util function `getCurrentTimestamp` that wraps the `Date` API. Then you can mock that function in test and control it more precisely.

If you want to keep it like this, that's fine too, but maybe add a comment explaining why

Reply:
4d26214f25eb45df7e0ed156a9c80af4454a232d, thanks for the suggestion!